chore: add README & LICENSE (#281)
klarna/electron-redux: README.md

When working with Electron, using Redux poses couple of problems, since main and renderer processes are isolated and IPC is a single way of communication between them. This library, enables you to register all your Redux stores in the main & renderer process, and enable cross-process action dispatching & _loose_ store synchronization.

![GitHub Workflow Status](https://img.shields.io/github/workflow/status/klarna/electron-redux/Release)
![npm (tag)](https://img.shields.io/npm/v/electron-redux/alpha)
![npm](https://img.shields.io/npm/dm/electron-redux)
![npm bundle size (version)](https://img.shields.io/bundlephobia/minzip/electron-redux/alpha)
[![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
[![Conventional Commits](https://img.shields.io/badge/Conventional%20Commits-1.0.0-yellow.svg)](https://conventionalcommits.org)

## Installation

```sh
# with yarn
yarn add electron-redux@alpha

# with npm
npm install electron-redux@alpha
```

For more details, please see [the Installation docs page](#todo).

## Documentation

> electron-redux docs are located at **electron-redux.js.org**. You can find there:

Reviewer: Can we discuss whether we can perhaps get a klarna.com URL?

-   [Getting started](#todo)
-   [Recipes](#todo)
-   [API Reference](#todo)

## Quick start

> electron-redux comes as a [Redux StoreEnhancer](https://redux.js.org/understanding/thinking-in-redux/glossary#store-enhancer). To initialize your stores using it, you just need decorate your stores in the `main` and `renderer` process of electron with provided enhancers:

Reviewer: ```suggestion
electron-redux comes as a [Redux StoreEnhancer](https://redux.js.org/understanding/thinking-in-redux/glossary#store-enhancer). To initialize your stores, you just need to decorate them in the `main` and `renderer` processes of electron with their respective enhancers:
```

```ts
// main.ts
import { mainStateSyncEnhancer } from 'electron-redux'

const store = createStore(reducer, mainStateSyncEnhancer())
```

```ts
// renderer.ts
import { rendererStateSyncEnhancer } from 'electron-redux'

const store = createStore(reducer, rendererStateSyncEnhancer())
```

That's it!

> You are now ready to fire actions in any of the stores / processes, and depending on the [scope](#scoped-actions) the main store will broadcast them to correct processes.

Reviewer: ```suggestion
You are now ready to fire actions from any of your processes, and depending on the [scope](#scoped-actions) the main store will broadcast them across your application.
```

Please check out the [docs](#todo) for more recipes and examples!

## Actions

> Actions fired **MUST be [FSA](https://github.com/acdlite/flux-standard-action#example)-compliant**, i.e. have a type and payload property. Any actions not passing this test will be ignored and simply passed through to the next middleware.

Reviewer: ```suggestion
Actions fired **MUST be [FSA](https://github.com/acdlite/flux-standard-action#example)-compliant**, i.e. have a `type` and `payload` property. Any actions not passing this test will be ignored and simply passed through to the next middleware.
```

Reviewer: > to the next middleware

Is this still correct since we're using an enhancer instead of middleware?

> Nota bene, `redux-thunk` is not FSA-compliant out of the box, but can still produce compatible actions once the async action fires.

Furthermore, actions (and that includes payloads) **MUST be serializable**.

> You can extend default JSON serialization used, by [providing your own serialization/deserialization functions](#todo).

### Scoped actions

> By default, all actions are broadcasted to all registered stores. However, some state should only live in the renderer (e.g. isPanelOpen). electron-redux introduces the concept of action scopes.

Reviewer: ```suggestion
By default, all actions are broadcast to all registered stores. However, some state should only live in the renderer (e.g. `isPanelOpen`). electron-redux introduces the concept of action scopes.
```
